atlp-rwanda/eagles-ec-be
# #187419154-chore: Configure ESLint and Git Pre commit hook to run tests

#### What does this PR do?

This PR introduces ESLint for code linting and adds comprehensive tests to cover critical endpoints for home and user routes.

#### Description of Task to be completed?

- Integrate ESLint for consistent code quality.
- Develop tests to cover endpoints for home ("/") and user ("/api/v1/users") routes.

#### How should this be manually tested?

1. Ensure ESLint is correctly configured and linting is enforced once u run test it will lint app first. 
2. <img width="611" alt="Screenshot 2024-04-13 at 16 39 29" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/9463053c-958a-49e0-ba34-df27997f4d08">
3.  ### to fix linting error before u jump to test u must run `npm run lint:fix ` to fix linting error for u
4. Run `npm test` to execute the test suite and verify endpoint coverage and functionality.

#### Any background context?

This PR enhances code quality by enforcing ESLint rules and ensures critical endpoints are thoroughly tested for functionality and reliability.

#### Relevant pivotal tracke id

[#187419154](https://www.pivotaltracker.com/story/show/187419154)


## screenshots


## when u try to run test while u have some lint error u will get linting error like this 
<img width="780" alt="Screenshot 2024-04-13 at 16 42 29" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/2704c702-860c-4dd7-9858-8eb21f3a5712">

## while if u don't have lint error u will see this test pass result alongside code coverage as below
<img width="905" alt="Screenshot 2024-04-13 at 16 38 48" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/3568af1d-ec75-4b83-a7f8-5acb17b4cf1a">
